desc/geometry/surface.py

```diff
@@ -890,3 +893,315 @@ def set_coeffs(self, l, m=0, R=None, Z=None):
             if ZZ is not None:
                 idxZ = self.Z_basis.get_idx(ll, mm, 0)
                 self.Z_lmn = put(self.Z_lmn, idxZ, ZZ)
+
+
+class PoincareSurface(ZernikeRZToroidalSection):
```

Comment:
im still not convinced we need a separate class for this vs just adding lambda to the existing cross section object. would like to hear other's opinions

Reply:
Personally, I think any cross-section object should have lambda in it. But I don't know the other use cases of this class. What do you think @dpanici @ddudt ?

Reply:
I'm pretty sure we don't use the base `ZernikeRZToroidalSection` anywhere right now, it was added for use with the poincare stuff so if we need to modify it for that it should be fine

Reply:
Ok, with the latest changes, I merged the PoincareSurface and ZernikeRZToroidaSection classes into new one called PoincareRZLSection (the naming is to specify the basis)